share/spack/setup-env.sh

```diff
@@ -89,6 +89,43 @@ function spack {
             fi
             return
             ;;
+        "env")
+            _sp_arg=""
+            if [ -n "$1" ]; then
+                _sp_arg="$1"
+                shift
+            fi
+
+            if [ "$_sp_arg" = "-h" ]; then
+                command spack env -h
+            else
+                case $_sp_arg in
+                    activate)
+                        _a="$@"
+                        if [ -z "$1" -o "${_a#*--sh}" != "$_a" -o "${_a#*--csh}" != "$_a" ]; then
```

Comment:
@tgamblin this branch should also be taken if the args include `-h`, otherwise, the help will be printed out and passed to `eval` in the `else` branch of this